src/pvecontrol/node.py

```diff
@@ -51,30 +52,26 @@ def __str__(self):
     def _init_vms(self):
         self.vms = []
         if self.status == NodeStatus.ONLINE:
-            self.vms = [
-                PVEVm(self._api, self.node, vm["vmid"], vm["status"], vm)
-                for vm in self._api.nodes(self.node).qemu.get()
-            ]
+            self.vms = [PVEVm(self._api, self.node, vm["vmid"], vm["status"], vm) for vm in self.get_resources_vms()]
 
     def _init_allocatedmem(self):
         """Compute the amount of memory allocated to running VMs"""
         self.allocatedmem = 0
         for vm in self.vms:
             if vm.status != VmStatus.RUNNING:
                 continue
-            # This is in MB in configuration
-            self.allocatedmem += int(vm.config["memory"]) * 1024 * 1024
+            self.allocatedmem += vm.maxmem
 
     def _init_allocatedcpu(self):
         """Compute the amount of cpu allocated to running VMs"""
         self.allocatedcpu = 0
         for vm in self.vms:
             if vm.status != VmStatus.RUNNING:
                 continue
-            if "sockets" in vm.config:
-                self.allocatedcpu += vm.config["sockets"] * vm.config["cores"]
-            else:
-                self.allocatedcpu += vm.config["cores"]
+            self.allocatedcpu += vm.cpus
+
+    def get_resources_vms(self):
```

Comment:
If I understand, you prefere to remove `_init_vms` to have `vm` property to load PVEVm from resources ?

Reply:
No, I just mean that it could be:

```python
@property
def resources_vms(self):
    return [resource for resource in self.resources if resource["type"] == "qemu"]
```